Make isort & black friends

mathfac opened the pull request:
## Motivation and Context
- Added `pre-commit` hooks for style checking
- Improved `isort` and `black` compatibility: added `toml/pyproject` support to isort pre-commit and added [an option recommended for compatibility](https://github.com/psf/black#how-black-wraps-lines)
- Added [seed-isort-config](https://github.com/asottile/seed-isort-config) pre-commit hook to manage third parties list
- Updated CI to use `pyproject` config for `isort` and `black`
- Moved `isort` config from `pyproject` to `.editorconfig`
- Added `ensure_newline_before_comments=true` to `isort` to be compatible with `black`. Will take action with next version of `isort`.
- Added habitat packages to `known_first_party`

<!--- Why is this change required? What problem does it solve? -->
<!--- Please link to an existing issue here if one exists. -->
<!--- (we recommend to have an existing issue for each pull request) -->

## How Has This Been Tested
No conflicts with style checking.

<!--- Please describe here how your modifications have been tested. -->

## Types of changes

<!--- What types of changes does your code introduce? Leave all the items that apply: -->
- Docs change / refactoring / dependency upgrade
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

apsdehal commented on pyproject.toml:
```diff
@@ -0,0 +1,28 @@
+
+[tool.black]
+line_length = 79
+exclude = '''
```
Is there a possibility of picking this up from `.gitignore`?

mathfac replied:
black doesn't support [that feature yet](https://github.com/mdagaro/isort/commit/af012ef18e4680f2b3780d36bca9e3f25fa19a60), but it should run on files that are part of repo on `pre-commit` hook.

apsdehal commented on pyproject.toml:
```diff
@@ -0,0 +1,28 @@
+[tool.isort]
```
There is a known_first_party option too.

apsdehal commented on pyproject.toml:
```diff
@@ -0,0 +1,28 @@
+[tool.isort]
+known_third_party = ["PIL", "attr", "cv2", "gym", "habitat_sim", "imageio", "matplotlib", "numba", "numpy", "orbslam2", "pytest", "quaternion", "requests", "scipy", "setuptools", "torch", "torchvision", "tqdm", "version", "yacs"]
+multi_line_output = 3
```
Isort can also pick up .editorconfig setting, so maybe reuse that?

apsdehal commented on .pre-commit-config.yaml:
```diff
@@ -0,0 +1,29 @@
+    rev: v2.2.3
+    hooks:
+    -   id: trailing-whitespace
+    -   id: check-added-large-files
```
Great!!

erikwijmans commented on .pre-commit-config.yaml:
```diff
@@ -0,0 +1,29 @@
+    rev: v4.3.20
+    hooks:
+    -   id: isort
+        language_version: python3.6
```
I don't believe these are language versions are needed and they prevent you from using python3.7

mathfac commented:
Addressed the comments:
- Moved `isort` config from `pyproject` to `.editorconfig`
- Added `ensure_newline_before_comments=true` to `isort` to be compatible with `black`. Will take action with next version of `isort`.
- Added habitat packages to `known_first_party`
